src/main/java/org/folio/circulation/resources/RequestFromRepresentationService.java

```diff
@@ -179,22 +190,36 @@
       }));
   }
 
+  private Item getFirstAvailableItem(Request request) {
+    return request.getItemsOfInstance() == null ? null :
+      request.getItemsOfInstance().stream()
+        .filter(Objects::nonNull)
+        .filter(item -> ItemStatus.AVAILABLE.equals(item.getStatus()))
+        .findFirst()
+        .orElse(null);
+  }
+
   private CompletableFuture<Result<Request>> fetchItemAndLoanForRecallTlrRequest(Request request) {
     if (errorHandler.hasAny(INVALID_INSTANCE_ID)) {
       return ofAsync(() -> request);
     }
 
-    return itemByInstanceIdFinder.getItemsByInstanceId(UUID.fromString(request.getInstanceId()))
+    return completedFuture(succeeded(request.getItemsOfInstance()))
       .thenComposeAsync(r -> r.after(items -> loanRepository.findLoanWithClosestDueDate(
         mapToItemIds(items)))
         .thenApply(resultLoan -> resultLoan.map(request::withLoan))
         .thenComposeAsync(requestResult -> requestResult.combineAfter(
-          record -> itemRepository.fetchFor(record.getLoan()), Request::withItem))
+          r2 -> itemRepository.fetchFor(r2.getLoan()), Request::withItem))
         .thenComposeAsync(requestResult -> requestResult.combineAfter(
           this::getUserForExistingLoan, this::addUserToLoan)))
       .thenApply(r -> errorHandler.handleValidationResult(r, INSTANCE_DOES_NOT_EXIST, request));
   }
 
+  private CompletableFuture<Result<Request>> fetchItemsOfInstanceForTheRequest(Request request) {
+    return itemByInstanceIdFinder.getItemsByInstanceId(UUID.fromString(request.getInstanceId()))
```

Comment:
NPE if `request.getInstanceId()` is null

Reply:
we check request.getInstanceId() to null above, so that, no needs to be checked here